sway/commands.c

```diff
@@ -143,44 +143,34 @@ static int handler_compare(const void *_a, const void *_b) {
 	return strcasecmp(a->command, b->command);
 }
 
-struct cmd_handler *find_handler(char *line, struct cmd_handler *cmd_handlers,
-		int handlers_size) {
-	struct cmd_handler d = { .command=line };
-	struct cmd_handler *res = NULL;
-	sway_log(SWAY_DEBUG, "find_handler(%s)", line);
-
-	bool config_loading = config->reading || !config->active;
-
-	if (!config_loading) {
-		res = bsearch(&d, command_handlers,
-				sizeof(command_handlers) / sizeof(struct cmd_handler),
-				sizeof(struct cmd_handler), handler_compare);
-
-		if (res) {
-			return res;
-		}
+struct cmd_handler *find_handler(char *line, struct cmd_handler *handlers,
+		size_t handlers_size) {
+	if (!handlers || !handlers_size) {
+		return NULL;
 	}
+	struct cmd_handler query = { .command = line };
+	return bsearch(&query, handlers,
+			handlers_size / sizeof(struct cmd_handler),
+			sizeof(struct cmd_handler), handler_compare);
+}
 
+static struct cmd_handler *find_handler_ex(char *line,
```

Comment:
This is used by only one function. Would it be used by more than one in the future?

Reply:
I'm thinking about changing `config_subcommand` to use this, which would simplify the logic in the command handlers with subcommands (or the ones that have config/command only handlers anyway).